### PR TITLE
Total Scattering :: Add most basic form of total scattering workflow

### LINKED ISSUE #21294: Total scattering :: Add minimum case for Polaris to ISIS_Powder
A very basic form of PDF analysis for total scattering data for a single bank can be achieved by loading the sample, converting to Q and running `PDFFourierTransform`.

As a base case that can be built on for total scattering. This issue will cover adding this functionality to isis_powder POLARIS instrument. 

This should be run by setting up the polaris instrument object, and executing `polaris.totalscatter()` after running `create_van` and `focus`

### PR DESCRIPTION
**Pull request by ElliotAOram**
* Added the most basic case for total scattering to the POLARIS instrument in the isis_powder diffraction scripts. 
* No real scientific meaning can be obtained from this case, but it should make a good foundation to build on moving forward (adding additional corrections + merging banks)
* To avoid continually updating a reference file, a system test has been added that measures peak intensity for the first peak of interest (~4 Angstrom).
* Updated POLARIS documentation with total scattering functionality

**To test:**
* Open Mantid plot
* Point your data search directory to: `\\olympic\babylon5\Public\ElliotOram\polaris` **(make sure this is the first data search directory in the list)**
* Also point the data search directory to the reference files directory `mantid\ExternalData\Testing\SystemTests\tests\analysis\reference`
* Run the following script
```
from isis_powder import Polaris
DIRS = config['datasearch.directories'].split(';')
# Relative to system data folder
working_folder_name = "polaris"


# Relative to input folder
calibration_folder_name = "calibration"
calibration_map_rel_path = "polaris_mapping.yaml"
config_rel_path = "Master_copy_of_grouping_file_with_essential_masks.cal"

# Generate paths
input_dir = os.path.join(DIRS[0], 'input')
output_dir = os.path.join(DIRS[0], 'output')

calibration_map_path = os.path.join(input_dir, calibration_map_rel_path)
calibration_dir = os.path.join(input_dir, calibration_folder_name)
config_path =os.path.join(calibration_dir, config_rel_path)

inst_obj = Polaris(user_name="Mantid-Test", calibration_mapping_file=calibration_map_path,
                   calibration_directory=calibration_dir, output_directory=output_dir, mode="PDF")

inst_obj.create_total_scattering_pdf(run_number='97947', merge_banks=False)
```
* A new group workspace (`pdf_output`) containing 5 workspaces should be present in the ADS
* These spectra should be Y unit: `PDF`,  X unit: `AtomicDistance`
* Plotting all banks together should look like this (zoomed in to avoid the majority of the initial tail):
image

* Try changing merge_banks to `True` (this should throw an exception as it is not yet implemented)

* Try loading in the focused data file to mantidplot and naming it either `97947-Results-TOF-Grp` or `97947-Results-D-Grp` then re-running the above script but changing the output directory to some other path (anywhere will do!) - this should ensure that the focused workspaces can be found using the ADS rather than only by file.
  * you could also do the ADS check with the output directory as shown in the script originally. In this case, ensure that no data is loaded and the workspaces come from the ADS rather than reloading the data - you should be able to assert this from looking at the result log.

Fixes #21294

**Release Notes** 
Under the total scattering section [here](https://github.com/mantidproject/mantid/blob/6c43079e50c9d0d36eb7a067388e6e7863360c0f/docs/source/release/v3.12.0/diffraction.rst)

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by josephframsay — CHANGES_REQUESTED** [review]
The changes mostly look quite nice (and it's quite cool to see other things getting put into `isis_powder`!). Unfortunately when I run the script on my machine, I appear to get very wide bins, unlike the image you attached. Could this be something to do with the recently updated Polaris system test data, or is something else going on?
pdf

File: scripts/Diffraction/isis_powder/polaris_routines/polaris_algs.py [hunk 1]
```diff
@@ -76,6 +76,15 @@ def save_unsplined_vanadium(vanadium_ws, output_path):
     mantid.DeleteWorkspace(converted_group)
 
 
+def generate_ts_pdf(focused_ws, merge_banks=False):
+    pdf_output = mantid.ConvertUnits(InputWorkspace=focused_ws, Target="MomentumTransfer")
+    if merge_banks:
+        raise RuntimeError("Merging banks is currently not supported")
```
[the remark this answers is not in the document]
Comment (ElliotAOram) [reply scripts/Diffraction/isis_powder/polaris_routines/polaris_algs.py]: Yes you're right. 
Currently if you merge banks (merge_banks=True) then it should fail. Only the non-merge_banks version is implemented. I'll update the test message

File: Testing/SystemTests/tests/analysis/ISIS_PowderPolarisTest.py [hunk 2]
```diff
@@ -95,6 +95,34 @@ def cleanup(self):
...
+                                [11, 0.285],
+                                [23, 0.504],
+                                [37, 0.382],
+                                [52, 0.700]]
+        for index, ws in enumerate(self.pdf_output):
+            self.assertAlmostEqual(ws.dataY(0)[expected_peak_values[index][0]], expected_peak_values[index][1],
+                                   places=3)
+
+
+def run_total_scattering(focused_ws, merge_banks):
```
[the remark this answers is not in the document]
Comment (ElliotAOram) [reply Testing/SystemTests/tests/analysis/ISIS_PowderPolarisTest.py]: I agree that it makes sense to keep this consistent with the rest of `isis_powder` so I'll make the change.

**Comment by ElliotAOram** [comment]
@josephframsay I've just pushed the changes to make the ts function more `isis_powder`-like. I'll have a look into the output issue next and let you know when it's ready to test again!

**Comment by ElliotAOram** [comment]
@josephframsay this should be good to go now. The test failures seem unrelated, but I'll rebuild them just in case. Also note that the test instructions have been updated since last time.

**Comment by josephframsay** [comment]
@ElliotAOram The new changes look pretty nice - I like your way of trying the ADS first for the focused run (I presume your testing instructions should say rename to `97947-Results-TOF-Grp`, not `97947-Result-TOF-Grp`?). Unfortunately `pdf_output` still appears to have really wide bins - the plot is the same as it was before

File: docs/source/techniques/ISISPowder-Polaris-v1.rst [hunk 3]
```diff
@@ -176,14 +176,25 @@ from focused POLARIS data, with a view performing further total scattering analy
 With no merging criteria specified, *merge_banks=False* a PDF will be generated for each bank within
 the focused_workspace.
 
+This function requires the run_number you which to analyse. The focused file for this run number must
```
Comment (josephframsay) [inline docs/source/techniques/ISISPowder-Polaris-v1.rst]: Nit: presumably should be "run_number you **wish** to analyse"?

**Comment by ElliotAOram** [comment]
@josephframsay yes please I'll take a look at the file. At the moment, I am expecting there to be a mismatch in the binning between spectra - but I'll take a look

File: docs/source/techniques/ISISPowder-Polaris-v1.rst (hunk 3 above)
Comment (ElliotAOram) [reply docs/source/techniques/ISISPowder-Polaris-v1.rst]: Yep good spot

**Review by josephframsay — APPROVED** [review]

**Comment by josephframsay** [comment]
We clarified the difference in binning (just a matter of plotting the wrong section of the X axis), so everything looks good!

**Comment by ElliotAOram** [comment]
System test failure appears unrelated

**Pull request merged**